src/main/webapp/js/utils/inputs/date-picker.jsx

```diff
@@ -2,13 +2,14 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import ReactDatePicker from 'react-datepicker';
 import 'react-datepicker/dist/react-datepicker.css';
-import { parseISO } from 'date-fns';
+import { parseISO, format } from 'date-fns';
 
 const DatePicker = ({
-  onChange, value, dateFormat, ...props
+  onChange, value, dateFormat, showTimeSelect, ...props
 }) => {
   const handleChange = (date) => {
-    onChange(date);
+    const valDate = showTimeSelect ? date.toISOString() : format(date, dateFormat);
```

Comment:
you should check if the the type is string first and just use this value if it is. When showTimeSelect is false use "YYYY-MM-DD" instead dateFormat